sklearn/inspection/_partial_dependence.py

```diff
@@ -137,27 +140,33 @@ def _partial_dependence_brute(est, grid, features, X, response_method):
 
     for new_values in grid:
         X_eval = X.copy()
-        for i, variable in enumerate(features):
+        for i, variable in enumerate(features_indices):
             if hasattr(X_eval, 'iloc'):
                 X_eval.iloc[:, variable] = new_values[i]
             else:
                 X_eval[:, variable] = new_values[i]
 
         try:
-            predictions = prediction_method(X_eval)
+            # Note: predictions is of shape
+            # (n_points,) for non-multioutput regressors
+            # (n_points, n_tasks) for multioutput regressors
+            # (n_points, 1) for the regressors in cross_decomposition (I think)
+            # (n_points, 2) for binary classification
+            # (n_points, n_classes) for multiclass classification
+            predictions.append(prediction_method(X_eval))
         except NotFittedError:
             raise ValueError(
                 "'estimator' parameter must be a fitted estimator")
 
-        # Note: predictions is of shape
-        # (n_points,) for non-multioutput regressors
-        # (n_points, n_tasks) for multioutput regressors
-        # (n_points, 1) for the regressors in cross_decomposition (I think)
-        # (n_points, 2) for binary classification
-        # (n_points, n_classes) for multiclass classification
+    return predictions
+
 
+def _partial_dependence_brute(est, grid, features, X, response_method):
+    averaged_predictions = []
+    predictions = _get_predictions(est, grid, features, X, response_method)
+    for prediction in predictions:
         # average over samples
-        averaged_predictions.append(np.mean(predictions, axis=0))
+        averaged_predictions.append(np.mean(prediction, axis=0))
```

Comment:
```suggestion
        averaged_predictions.append(np.mean(pred, axis=0))
```

Reply:
Added in Added in https://github.com/scikit-learn/scikit-learn/pull/16164/commits/384d6b7f6822b431894aa776a4892f7829eb50dc